key_derivation_test.go

```diff
@@ -13,24 +15,32 @@ func TestValidSessionKeys(t *testing.T) {
 	expectedSessionSalt := []byte{0x30, 0xCB, 0xBC, 0x08, 0x86, 0x3D, 0x8C, 0x85, 0xD4, 0x9D, 0xB3, 0x4A, 0x9A, 0xE1}
 	expectedSessionAuthTag := []byte{0xCE, 0xBE, 0x32, 0x1F, 0x6F, 0xF7, 0x71, 0x6B, 0x6F, 0xD4, 0xAB, 0x49, 0xAF, 0x25, 0x6A, 0x15, 0x6D, 0x38, 0xBA, 0xA4}
 
-	sessionKey, err := generateSessionKey(labelSRTPEncryption, masterKey, masterSalt)
+	sessionKey, err := aesCmKeyDerivation(labelSRTPEncryption, masterKey, masterSalt, 0, len(masterKey))
 	if err != nil {
 		t.Errorf("generateSessionKey failed: %v", err)
 	} else if !bytes.Equal(sessionKey, expectedSessionKey) {
 		t.Errorf("Session Key % 02x does not match expected % 02x", sessionKey, expectedSessionKey)
 	}
 
-	sessionSalt, err := generateSessionSalt(labelSRTPSalt, masterKey, masterSalt)
+	sessionSalt, err := aesCmKeyDerivation(labelSRTPSalt, masterKey, masterSalt, 0, len(masterSalt))
 	if err != nil {
 		t.Errorf("generateSessionSalt failed: %v", err)
 	} else if !bytes.Equal(sessionSalt, expectedSessionSalt) {
 		t.Errorf("Session Salt % 02x does not match expected % 02x", sessionSalt, expectedSessionSalt)
 	}
 
-	sessionAuthTag, err := generateSessionAuthTag(labelSRTPAuthenticationTag, masterKey, masterSalt)
+	authKeyLen, err := ProtectionProfileAes128CmHmacSha1_80.authKeyLen()
+	assert.NoError(t, err)
+
+	sessionAuthTag, err := aesCmKeyDerivation(labelSRTPAuthenticationTag, masterKey, masterSalt, 0, authKeyLen)
 	if err != nil {
 		t.Errorf("generateSessionAuthTag failed: %v", err)
 	} else if !bytes.Equal(sessionAuthTag, expectedSessionAuthTag) {
 		t.Errorf("Session Auth Tag % 02x does not match expected % 02x", sessionAuthTag, expectedSessionAuthTag)
 	}
 }
+
+func TestIndexOverKDR(t *testing.T) {
+	_, err := aesCmKeyDerivation(labelSRTPAuthenticationTag, []byte{}, []byte{}, 1, 0)
+	assert.Error(t, err)
```

Comment:
It's not expected to return error, but currently not implemented yet. Better to add a comment.

Reply:
:+1: added comment.